### ISSUE TITLE
Feature request: add basicAuth for grafana_datasource

### ISSUE THREAD

**Issue by gerhard-strauss** [voxpupuli/puppet-grafana#43]
Hello,

Grafana Datasources can be "protected" with Basic Authentication, whose Credentials can be configured in the dashboard, but not with the grafana_datasource Provider.
```
"basicAuth" : true,
"basicAuthUser" : "foo",
"basicAuthPassword" : "bar"
```
In Grafana 4.2.0 is also an additional Parameter
```
"withCredentials" : true
```
It would be great if these Parameter could set with the grafana_datasource Provider. 
Another possible Solution would be an additional Parameter, which takes a hash with custom config.

**Comment by jpuskar** [comment]
I hacked this together for Elasticsearch.
https://github.com/jpuskar/puppet-grafana/pull/1

**Comment by gerhard-strauss** [comment]
I did a quick test and it worked.
But i use graphite as data type, so i changed it local to 
if data[:type].to_s == 'graphite'
to get it to work.

imho i think the best way would be to set basic_auth_user and basic_auth_password only when basic_auth is true, not dependant on the data source type.
but i asume you did this for testing purpose

**Comment by brandonrdn** [comment]
@bastelfreak Fixed with #86

**Comment by anarcat** [comment]
fixed in #86?
